run_tests.py

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env python
+
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import os
+import sys
+import time
+import urllib
+import tarfile
+from subprocess import check_call
+
+import start
+
+
+DIR_TEST_ENV = "test/venv"
+DIR_JENKINS_ENV = "jenkins-env"
+VERSION_VIRTUALENV = "12.0.5"
+
+os.environ["JENKINS_HOME"] = "jenkins-master"
+
+
+def check_patches():
+    print "Checking patches"
+    check_call(["./test/check_patches.sh"])
+
+
+class Jenkins(object):
+    def __init__(self):
+        if self._require_venv_setup():
+            print "Running setup"
+            check_call(["./setup.sh", DIR_JENKINS_ENV])
+        print "Starting Jenkins"
+        self.proc = start.start_jenkins()
+
+    def _require_venv_setup(self):
+        if os.path.exists(DIR_JENKINS_ENV) and not os.environ.get('CI'):
+            print "Jenkins environment already exists!"
+            while True:
+                user_input = raw_input("Would you like to recreate it? (y/n): ")
+                if user_input.lower().startswith("y"):
+                    break
+                elif user_input.lower().startswith("n"):
+                    return False
+        return True
+
+    def wait_for_started(self):
+        import requests
+        max_time = time.time() + 60
+        session = requests.Session()
+        while time.time() <= max_time:
+            try:
+                if session.get("http://localhost:8080").status_code == 200:
+                    return True
+            except requests.exceptions.ConnectionError:
+                pass
+            time.sleep(0.5)
+        return False
```

Comment:
Mh, maybe we should raise a timeout error here? It would make it easier for error handling.